Docs/source/install/hpc/narval.rst

```diff
@@ -0,0 +1,153 @@
+========================================================
+
+The `Narval cluster <https://www.calculquebec.ca/en/nouvelle/narval-a-new-supercomputer-dedicated-to-scientific-research/>`__ is located at McGill University.
+
+Introduction
+------------
+
+If you are new to this system, **please see the following resources**:
+
+* `Narval Wiki <https://docs.alliancecan.ca/wiki/Narval/en>`__
+* `Infrastructures and Services <https://www.calculquebec.ca/en/academic-research-services/infrastructures-services/#ouvert>`__
+
+Through Narval we have access to CPU and GPU nodes.
+
+Installation
+------------
+
+Narval has no WarpX preinstalled, we need to compile it from scratch. Since ``$HOME`` directory is designed for compiling codes, the following operations will be executed in ``$HOME`` directory.
+
+Download WarpX
+^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+    cd $HOME
+    git clone https://github.com/ECP-WarpX/WarpX.git
+
+Building WarpX
+^^^^^^^^^^^^^^
+
+* To build it with CPU backend, load the following module
+
+.. code-block:: bash
+
+    module load StdEnv/2023
+
+This environment contains ``gcc/12.3, openmpi/4.1.5`` and other usefule modules. Now build WarpX as described in `documentation <https://warpx.readthedocs.io/en/latest/install/cmake.html>`__
+
+.. code-block:: bash
+
+    cmake -S $HOME/WarpX -B $HOME/WarpX/build -DWarpX_DIMS="1;2;3;RZ" \
+        -DWarpX_COMPUTE=OMP \
+        -DWARX_MPI=ON \
+    cmake --build build -j 4
+
+
+* To build it with GPU backend, load the following modules
+
+.. code-block:: bash
+
+    module load StdEnv/2023 cuda/12.2
+
+The only thing we need to change is the ``-DWarpX_COMPUTE`` flag,
+
+.. code-block:: bash
+
+    cmake -S $HOME/WarpX -B $HOME/WarpX/build -DWarpX_DIMS="1;2;3;RZ" \
+        -DWarpX_COMPUTE=CUDA \
+        -DWarpX_MPI=ON \
+    cmake --build build -j 4
+
+Python Bindings
+^^^^^^^^^^^^^^^
+
+1. Create virtualenv
+
+.. code-block:: bash
+
+    virtualenv --system-site-packages $HOME/.venvs/warpx
+    source $HOME/.venvs/warpx/bin/activate
+
+2. Compile WarpX with extra flag ``-DWarpX_PYTHON=ON``, and install ``pywarpx`` into the virtual environment
+
+.. code-block:: bash
+
+    cmake -S $HOME/WarpX -B $HOME/WarpX/build -DWarpX_DIMS="1;2;3;RZ" \
+        -DWarpX_COMPUTE=CUDA \
+        -DWarpX_MPI=ON \
+        -DWarpX_PYTHON=ON
+    cmake --build build --target pip_install -j 4
+
+Running
+-------
+
+We are not allowed to submit jobs or run jobs interactively in ``$HOME`` directory, ``$SCRATCH`` directory is the correct place to do this.
+
+Testing and Debugging
+^^^^^^^^^^^^^^^^^^^^^
+
+To run the simulation interactively, we need to request some gpu
+resource first
+
+.. code-block:: bash
+
+    salloc --time=1:00:00 --mem=1G --gpus=1
+
+*  must set the mem to a number larger than the default one, other wise
+   ``out of memory error`` occurs.
+
+Now we can run the simulation in command line,
+
+.. code-block:: bash
+
+    module load <modules>
```

Comment:
When submitting jobs with slurm, interactive or non-interactive, the job will derive the environment it was submitted from. Thus, there is no need to reload the modules in the job script if one sources the `warx.profile` consistently on login.